lib/jsdom/browser/Window.js

```diff
@@ -1,13 +1,13 @@
 "use strict";
 
 var CSSStyleDeclaration = require("cssstyle").CSSStyleDeclaration;
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+var file = require("./file");
```

Comment:
Yeah, this is totally understandable, for classes for which the spec gives working constructors. We can't avoid creating a new constructor per-window/document for those. Seems like a good pattern.